applications/nucleo_io/targets/freertos.armv6m.st-stm32f091rc-nucleo-dev-board/main.cxx

```diff
@@ -337,6 +337,21 @@ constexpr const Gpio *const kPortDEGpio[] = {
 openlcb::MultiConfiguredConsumer portde_consumers(stack.node(), kPortDEGpio,
```

Comment:
This would need to be updated to not instantiate the port D consumers if there are portD snap switches operating.
That needs two changes:
- the kPortDEGpio needs to be shorter (fewer entries) based on ifdefs
- the number of repetitions defined in the config.hxx needs to change based on the ifdef.

Reply:
There are 12 snap switch entries, which suggest that we would use PortD.0-7 and PortE.0-3 for snaps, when enabled, while PortE.4-7 is always standard constant on outputs. I've tweaked this to hopefully give us the consumers for "standard" use and not give us those consumers when we are using snaps.
Please review that code and let me know your thoughts.

Reply:
In an offline conversation, we have settled on 8 Snap outputs, using PortD. PortE will be standard, non-pulsed, outputs. The fixes for this have been pushed to my master fork and included in this pull request.